keps/sig-network/4858-ip-cidr-validation/README.md

```diff
@@ -0,0 +1,886 @@
+
+    - "interface addresses" (or "ifaddrs"), individual IP addresses
+      that are assigned to a particular network interface and thus
+      associated with the subnet attached to that interface (e.g.,
+      "`192.168.1.5/24`", meaning "the IP address `192.168.1.5`, which
+      is on the same network segment as the rest of `192.168.1.0/24`).
+
+  Although the latter type of CIDR string (where there are bits set in
+  the IP address beyond the "prefix length") is often used by CNI
+  plugins, it is not currently used in any Kubernetes API object.
+  Nonethless, `net.ParseCIDR` and `netip.ParsePrefix` accept both
+  types, and so we currently consider the "ifaddr" form to be valid in
+  fields where we are looking for a subnet (e.g.
+  `node.status.podCIDRs`) or a mask (e.g., a NetworkPolicy `ipBlock`).
+  If a string such as "`192.168.1.5/24`" is accepted by Kubernetes and
+  then passed unmodified to another API, it is not always clear
+  whether that API will end up treating it as the equivalent of
+  `192.168.1.0/24` or `192.168.1.5/32`, so again, this could result in
+  problems.
+
+- Though not a problem with _existing_ validation, it is also
+  important to note that the new `netip.ParseAddr` function accepts
+  addresses with "zone identifiers" attached, such as
+  "`fe80::1234%eth0`", meaning "the link-local address `fe80::1234` on
+  the network attached to `eth0` (as opposed to the same link-local
+  address on some other interface)". While specifying zone identifiers
+  is important in some contexts, it should not be necessary for any
+  existing Kubernetes API objects, and would confuse any code that
+  tries to use `netutils.ParseIPSloppy`, so we need to be careful to
+  not accept them.
+
+We will update all in-tree API types to use appropriate
+`utilvalidation` methods for IP/CIDR validation, and update the
+validation functions to allow only unambiguous values.
+
+The fields whose validation will be updated are:
+
+  - in `core`:
+    - `endpoints.subsets[].addresses[].ip`
+    - `endpoints.subsets[].notReadyAddresses[].ip`
+    - `node.spec.podCIDRs[]`
+    - `pod.spec.dnsConfig.nameservers[]`
+    - `pod.spec.hostAliases[].ip`
+    - `pod.spec.hostIP`
+    - `pod.spec.hostIPs[]`
+    - `pod.spec.podIP`
+    - `pod.spec.podIPs[]`
+    - `service.spec.clusterIP`
+    - `service.spec.clusterIPs[]`
+    - `service.spec.externalIPs[]`
+    - `service.spec.loadBalancerSourceRanges[]`
+    - `service.status.loadBalancer.ingress[].ip`
+
+  - in `networking`:
+    - `ingress.status.loadBalancer.ingress[].ip`
+    - `networkpolicy.spec.egress[].to[].ipBlock.cidr`
+    - `networkpolicy.spec.egress[].to[].ipBlock.except[]`
+    - `networkpolicy.spec.ingress[].from[].ipBlock.cidr`
+    - `networkpolicy.spec.ingress[].from[].ipBlock.except[]`
+    - `serviceCIDR.spec.cidrs[]`
+
+  - in `discovery`:
+    - `endpointslice.endpoints[].addresses[]`
+
+#### Updates to pre-existing objects with pre-existing invalid fields
```

Comment:
I think we just want ratcheting validation, https://github.com/kubernetes/enhancements/tree/master/keps/sig-api-machinery/4008-crd-ratcheting , 

> allows updates to resources that fail validation to succeed if the validation errors were on unchanged keypaths.

Reply:
That's for CRDs; this is just implementing the same functionality for (specific fields of) apiserver types.

Reply:
yeah, trying to highlight the logic applied there, so we apply the same, instead of special case it for the different types and fields with ip addresses